src/daemon/daemon.cpp

```diff
@@ -1168,7 +1168,7 @@ std::unordered_map<std::string, std::unordered_set<std::string>> map_snapshots_t
     for (const auto& it : instances)
     {
         if (it.snapshot_name().empty())
-            instance_snapshots_map[it.instance_name()];
+            instance_snapshots_map[it.instance_name()].clear();
```

Comment:
Using clear to make it obvious that we're removing anything that is there already.